### PR TITLE
[Routing] Fixed the importing of files using glob patterns that match multiple resources

### PR DESCRIPTION
**Pull request by skalpa**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #22160
| License       | MIT
| Doc PR        | n/a

This fixes the import of resources specified using glob patterns in `XmlFileLoader` and `YamlFileLoader`.

@nicolas-grekas This supersedes #25633 that's been in limbo since December despite your comments, so I decided to take care of it as I need this to work. I took care of the two loaders that are affected, and added tests.

### CODE REVIEW

**Comment by skalpa** [comment]
Failures are unrelated.

Also, I didn't know which branch to target: the source of the problem is a change in `Symfony\Component\Config\Loader\FileLoader` on the 3.3 branch, however in theory one could use  v3.3+ of the config component with an older version of the routing component... so I left it against 3.4 for now but can rebase to an older branch if you tell me which one.

**Comment by nicolas-grekas** [comment]
@skalpa thank you. Branch 3.4 is the good one. I think `RoutingConfigurator` also need a similar update (but not exact same logic: the `$subCollection` need to contain all the routes before passing it to the `ImportConfigurator` instance, you'll see.

**Comment by skalpa** [comment]
@nicolas-grekas I took care of `RoutingConfigurator` too, thanks for pointing that out.

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Routing/Tests/Fixtures/glob/php_dsl_bar.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Symfony\Component\Routing\Loader\Configurator;
+
+return function (RoutingConfigurator $routes) {
+    $collection = $routes->collection();
+
+    $collection->add('bar_route', '/bar')
+        ->defaults(['_controller' => 'AppBundle:Bar:view']);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/Tests/Fixtures/glob/php_dsl_bar.php]: `array()` instead of `[]` (same below, see fabbot patch)

Comment (skalpa) [reply src/Symfony/Component/Routing/Tests/Fixtures/glob/php_dsl_bar.php]: Sorry, old habits. Fixed.

**Comment by nicolas-grekas** [comment]
Thank you @skalpa.

**Pull request closed without merging**